hub/api/tests/test_dataset.py

```diff
@@ -57,3 +57,11 @@ def test_dataset_empty_load():
         ds_overwrite_empty = hub.load(path, overwrite=True)
         assert len(ds_overwrite_empty) == 0
         assert len(ds_overwrite_empty.tensors) == 0
+
+
+def test_public(hub_cloud_ds):
+    assert hub_cloud_ds.public
+    hub_cloud_ds.public = False
```

Comment:
I don't think this is the best api for changing privacy. make_public or make_private methods would look better in my opinion.